include/zephyr/net/net_if.h

```diff
@@ -2163,6 +2169,25 @@ static inline bool net_if_is_up(struct net_if *iface)
 	       net_if_flag_is_set(iface, NET_IF_RUNNING);
 }
 
+/**
+ * @brief Connect interface (if supported)
+ *
+ * @param iface Pointer to network interface
+ *
+ * @return 0 on success, -ESHUTDOWN if the iface is not admin-up, implementation-specific status
```

Comment:
I believe that we also have `ENOTSUPP` flag defined. Could be also included as an expected result code.